etna/models/nn/tft_native/tft.py

```diff
@@ -0,0 +1,653 @@
+
+    def _transform_features(self, x: TFTNativeBatch):
+        """Apply embedding layer to categorical input features and linear transformation to continuous features.
+
+        Parameters
+        ----------
+        x:
+            batch of data
+
+        Returns
+        -------
+        :
+            transformed batch of data
+        """
+        # Apply transformation to static data
+        for feature in self.static_reals:
+            x["static_reals"][feature] = self.static_scalers[feature](
+                x["static_reals"][feature].float()
+            )  # (batch_size, 1, hidden_size)
+        for feature in self.static_categoricals:
+            x["static_categoricals"][feature] = self.static_embeddings[feature](
+                x["static_categoricals"][feature].float()
+            )  # (batch_size, 1, hidden_size)
+
+        # Apply transformation to time varying data
+        for feature in self.time_varying_categoricals_encoder:
+            x["time_varying_categoricals_encoder"][feature] = self.time_varying_embeddings_encoder[feature](
+                x["time_varying_categoricals_encoder"][feature].float()
+            )  # (batch_size, encoder_length, hidden_size)
+        for feature in self.time_varying_categoricals_decoder:
+            x["time_varying_categoricals_decoder"][feature] = self.time_varying_embeddings_decoder[feature](
+                x["time_varying_categoricals_decoder"][feature].float()
+            )  # (batch_size, decoder_length, hidden_size)
+
+        for feature in self.time_varying_reals_encoder:
+            x["time_varying_reals_encoder"][feature] = self.time_varying_scalers_encoder[feature](
+                x["time_varying_reals_encoder"][feature].float()
+            )  # (batch_size, encoder_length, hidden_size)
+        for feature in self.time_varying_reals_decoder:
+            x["time_varying_reals_decoder"][feature] = self.time_varying_scalers_decoder[feature](
+                x["time_varying_reals_decoder"][feature].float()
+            )  # (batch_size, decoder_length, hidden_size)
+        return x
+
+    def forward(self, x: TFTNativeBatch, *args, **kwargs):
+        """Forward pass.
+
+        Parameters
+        ----------
+        x:
+            batch of data
+
+        Returns
+        -------
+        :
+            forecast with shape (batch_size, decoder_length, 1)
+        """
+        target_true = x["decoder_target"].float()  # (batch_size, decoder_length, 1)
+        batch_size = target_true.size()[0]
+        batch = self._transform_features(x)
+
+        #  Pass static data through variable selection and covariate encoder blocks
+        if self.num_static > 0:
+            static_features = batch["static_reals"].copy()
+            static_features.update(batch["static_categoricals"])
+            static_features = self.static_variable_selection(static_features)  # type: ignore
```

Comment:
Why do you need to ignore type here? I guess you can write sth like this to solve it:
```
if self. static_variable_selection is not None and  self.static_covariate_encoder is not None:
```